packages/web3-core/src/web3_config.ts

```diff
@@ -355,6 +356,13 @@ export abstract class Web3Config
 	}
 
 	public set defaultChain(val) {
+		if (
```

Comment:
Just a question,
Why do we need the changes in this file? Shouldn't we validate `hardfork` and  `basechain` between tx and defaultCommon. The changes won't let the user set a different value than the initial from what I understand.

Reply:
>Why do we need the changes in this file? Shouldn't we validate hardfork and basechain between tx and defaultCommon.
Yes, this was done in this [PR](https://github.com/web3/web3.js/pull/5419#discussion_r974377814).
This file change was to address this [comment](https://github.com/web3/web3.js/pull/5465#discussion_r981142284) 
so there is no confusion if there is any differences between common object and hardfork/basechain